src/components/keyboard/Keyboard.tsx

```diff
@@ -14,11 +14,21 @@ interface Props {
     style?: ViewStyle;
     height: number;
     focusedInput: string;
-    setValues: React.Dispatch<React.SetStateAction<any>>;
+    setValues: Dispatch<SetStateAction<any>>;
     onSubmit: () => void;
+    submitButtonTestID?: string;
```

Comment:
Are we sure we need that level of precision ? I am not quite sure we need to identify speicifically the submit or delete button as they won't relleay change

Reply:
What I mean is that we need the test ID to be able to isolate a component and check for modifications on it, for example a text changing or something like that. Those button I think will basically always (or almost) be a back arrow for delete and a submit icon for submit, so I'm not sure there is a need for QA to identify this sepcific part of the component to check any change in it.
But I might be missing a case :)

Reply:
Yes, i agree with you, but my idea for testing on the SmartApp was:
by finding submitButton by its testID, we  can simulate a press on it, and expect some actions to happen, for example navigating to next screen
```

await press('submitButtonTestID')
expect(navigation.navigate).toHaveBeenCalledWith('screenName')
```

Reply:
Hmm  good point. For the robot tests, we would need Stephanie to confirm whether or not she needs the ids to click on the buttons but she's on holidays this week. 
And for unit tests, I have no idea what we plan to do on the smartapp ^^'

Reply:
for now we wont do any unit tests on the smartapp, integration will cover the different interactions. But indeed, Stephanie will probably need testId for those tests

Reply:
@Vadim-x-squad why this PR has not been merged ?

Reply:
Not this I understand why you don't want unit test on the app itself @valentinmagrez